### PR TITLE
Add rspec environment for testing the API

### PR DESCRIPTION
**Pull request by elandesign**

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  config.expect_with :rspec do |expectations|
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+
+  config.mock_with :rspec do |mocks|
+    mocks.verify_partial_doubles = true
+  end
+
+  config.before :suite do
+    DatabaseCleaner[:sequel, {connection: ::DB}].strategy = :transaction
```
Comment (Haegin) [inline spec/spec_helper.rb]: Both transaction and truncation?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    DatabaseCleaner[:sequel, {connection: ::DB}].clean_with(:truncation)
```
Comment (Haegin) [inline spec/spec_helper.rb]: Could we extract `DatabaseCleaner[:sequel, {connection: ::DB}]` out? It's used 4 times here.

File: spec/api/lists_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+require "spec_helper"
+
+describe "lists" do
+  before do
+    List.new(name: "First List").save
```
Comment (Haegin) [inline spec/api/lists_spec.rb]: Is this hitting the DB? Do we want to worry about that being slow?

What about factories? I assume we'll add those when we need them?

Comment (elandesign) [inline spec/api/lists_spec.rb]: I figured this was simple enough that factories were an unnecessary overhead. Also, in-memory SQLite db should be pretty fast.

File: spec/spec_helper.rb (hunk 1 above)
Comment (elandesign) [inline spec/spec_helper.rb]: Good question. Cribbed from http://devblog.avdi.org/2014/07/08/zero-to-smoke-test-with-sinatra/, but will have a poke at it

File: spec/spec_helper.rb (hunk 2 above)
Comment (elandesign) [inline spec/spec_helper.rb]: Makes sense

**Pull request merged**